File: 560-SubarraySumEqualsK/560-SubarraySumEqualsK.cpp
// Last updated: 04/04/2026, 23:33:43
class Solution {
public:
    int subarraySum(vector<int>& nums, int k) {
        unordered_map<int,int> map;
        map[0]=1;
        int currSum = 0;
        int ans = 0;
        for(int num : nums) {
            currSum += num;
            if(map.find(currSum-k) != map.end()) {
                ans += map[currSum-k];
            }
            map[currSum]++;
        }
        return ans;
    }
};

File: 560-SubarraySumEqualsK/560-SubarraySumEqualsK.java
// Last updated: 04/04/2026, 23:33:48
class Solution {
    public int subarraySum(int[] nums, int k) {
        HashMap<Integer, Integer> map = new HashMap();
        int currSum = 0;
        int ans = 0;
        map.put(0, 1);
        for(int num : nums) {
            currSum += num;
            if(map.containsKey(currSum-k)) {
                ans += map.get(currSum-k);
            }
            map.put(currSum, map.getOrDefault(currSum, 0)+1);
        }
        return ans;
    }
}